src/oscar/apps/catalogue/abstract_models.py

```diff
@@ -1228,23 +1230,104 @@ class AbstractOption(models.Model):
     BOOLEAN = "boolean"
     DATE = "date"
 
+    SELECT = "select"
+    RADIO = "radio"
+    MULTI_SELECT = "multi_select"
+    CHECKBOX = "checkbox"
+
     TYPE_CHOICES = (
         (TEXT, _("Text")),
         (INTEGER, _("Integer")),
         (BOOLEAN, _("True / False")),
         (FLOAT, _("Float")),
         (DATE, _("Date")),
+        (SELECT, _("Select")),
+        (RADIO, _("Radio")),
+        (MULTI_SELECT, _("Multi select")),
+        (CHECKBOX, _("Checkbox")),
     )
 
+    empty_label = "------"
+    empty_radio_label = _("Skip this option")
+
     name = models.CharField(_("Name"), max_length=128, db_index=True)
     code = AutoSlugField(_("Code"), max_length=128, unique=True, populate_from='name')
     type = models.CharField(_("Type"), max_length=255, default=TEXT, choices=TYPE_CHOICES)
     required = models.BooleanField(_("Is this option required?"), default=False)
+    option_group = models.ForeignKey(
+        "catalogue.AttributeOptionGroup",
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+        related_name="product_options",
+        verbose_name=_("Option Group"),
+        help_text=_('Select an option group if using type "Option" or "Multi Option"'),
+    )
+    help_text = models.CharField(
+        verbose_name=_("Help text"), blank=True, null=True, max_length=255,
+        help_text=_("Help text shown to the user on the add to basket form")
+    )
+    order = models.IntegerField(
+        _("Ordering"),
+        null=True,
+        blank=True,
+        help_text=_("Controls the ordering of product options on product detail pages"),
+        db_index=True,
+    )
+
+    @property
+    def is_option(self):
+        return self.type in [self.SELECT, self.RADIO]
+
+    @property
+    def is_multi_option(self):
+        return self.type in [self.MULTI_SELECT, self.CHECKBOX]
+
+    @property
+    def is_select(self):
+        return self.type in [self.SELECT, self.MULTI_SELECT]
+
+    @property
+    def is_radio(self):
+        return self.type in [self.RADIO]
+
+    def add_empty_choice(self, choices):
+        if self.is_select and not self.is_multi_option:
+            choices = [("", self.empty_label)] + choices
+        elif self.is_radio:
+            choices = [(None, self.empty_radio_label)] + choices
+        return choices
+
+    def get_choices(self):
+        if self.option_group:
+            choices = [(opt.option, opt.option) for opt in self.option_group.options.all()]
+        else:
+            choices = []
+
+        if not self.required:
+            choices = self.add_empty_choice(choices)
+
+        return choices
+
+    def clean_value(self, value):
+        return json.dumps(value, cls=DjangoJSONEncoder)
+
+    def clean(self):
+        if self.type in [self.DATE, self.SELECT, self.MULTI_SELECT, self.CHECKBOX]:
```

Comment:
I had to add self.RADIO to this array and then the radio option seemed to work correctly when I manually tested it - 

this is my first github 'review' btw - hope I'm not stepping on anyones toes or reading the source incorrectly -

Reply:
ha yeah and self.DATE should not be there.

Reply:
good catch, self.DATE does not belong there, I guess I added the wrong value.